eng/test.yaml

```diff
@@ -1,44 +1,12 @@
 steps: 
 
-- task: DotNetCoreCLI@2
-  displayName: Test C# (dotnet test)
+- task: VSTest@2
   inputs:
-    command: 'custom'
-    projects: |
-      src/**/*.Tests.csproj
-    custom: 'test'
-    arguments: > 
-      --configuration $(_BuildConfig)
-      --collect:"Code Coverage"
-      --settings:CodeCoverage.runsettings
-      --filter "TestCategory!=PostDeployment&TestCategory!=Nightly&TestCategory!=PreDeployment"
-      --no-build
-  env:
-    NUGET_PACKAGES: '$(Build.SourcesDirectory)/.packages'
-  condition: succeededOrFailed()
-
-- task: Powershell@2
-  inputs: 
-    targetType: 'filePath'
-    filePath: eng\convert-codecoveragetoxml.ps1
-    arguments: -Path "$(system.defaultworkingdirectory)" -NugetPackagesPath "$(Build.SourcesDirectory)\.packages"
-  displayName: Convert Code Coverage to XML (powershell)
-
-- task: Palmmedia.reportgenerator.reportgenerator-build-release-task.reportgenerator@4
```

Comment:
Do you know if by refactoring this to use VSTest if it will actually upload the code coverage reports as they are today?

Reply:
These steps should be replaced. You can refactor whatever you need to in order to run the tests, but please ensure that the functionality to create and upload the formatted reports remain.

Reply:
Totally revamped the PR. Let's see how the new one goes.